ykbuild/build.rs

```diff
@@ -10,6 +17,17 @@ fn main() {
         return;
     }
 
+    // To avoid running cmake config/build/install, we only do anything with cmake if either a) no
+    // build of ykllvm exists b) the ykllvm submodule's hash has changed since we last built it.
+    //
+    // First we get the ykllvms submodule's hash.
+    let out = Command::new("git")
```

Comment:
What will this do when you have a working tree inside the ykllvm submodule? I think that changes in the working tree would change the ykllvm commit hash to be `<hash>-dirty`. However, would this mean that further edits to an already dirty ykllvm tree would not trigger a rebuild? In such cases, do we need to add a rerun-if-changed command?

Reply:
Sure, that makes sense.

Reply:
If the issue is still that `ykbuild` is both a build and normal dependency (cargo doesn't share deps between those), then [the solution I proposed earlier](https://github.com/ykjit/yk/pull/681#issuecomment-1544686556) may fix it naturally, only re-triggering a rebuild when a file changes.

I can try that tomorrrow, but I think this can go in in the meantime.

> at the moment this support is more aimed at "normal user of yk who doesn't know what to do with ykllvm"

I'd like to eventually get it so we can develop out of the submodule, as it would improve quality of life for developers too (they wouldn't have to worry about when and when not to rebuild ykllvm).

Reply:
> If the issue is still that ykbuild is both a build and normal dependency (cargo doesn't share deps between those), then https://github.com/ykjit/yk/pull/681#issuecomment-1544686556 may fix it naturally, only re-triggering a rebuild when a file changes.

Can't hurt to try!